src/java.base/share/classes/java/lang/package-info.java

```diff
@@ -25,35 +25,42 @@
 
 /**
  * Provides classes that are fundamental to the design of the Java
- * programming language. The most important classes are {@code
- * Object}, which is the root of the class hierarchy, and {@code
+ * programming language. The most important classes are {@link
+ * Object}, which is the root of the class hierarchy, and {@link
  * Class}, instances of which represent classes at run time.
  *
  * <p>Frequently it is necessary to represent a value of primitive
- * type as if it were an object. The wrapper classes {@code Boolean},
- * {@code Character}, {@code Integer}, {@code Long}, {@code Float},
- * and {@code Double} serve this purpose.  An object of type {@code
- * Double}, for example, contains a field whose type is double,
+ * type as if it were an object.<a id="wrapperClasses"></a>
+ * The <dfn>{@index "wrapper classes"}</dfn> {@link Boolean},
+ * {@link Character}, {@link Short}, {@link Integer}, {@link Long}, {@link Float},
+ * and {@link Double} serve this purpose.  An object of type {@code
+ * Double}, for example, contains a field whose type is {@code double},
  * representing that value in such a way that a reference to it can be
- * stored in a variable of reference type.  These classes also provide
+ * stored in a variable of reference type.
+ * As discussed in <cite>The Java Language Specification</cite>, the
+ * wrapper classes are involved in boxing (JLS {@jls 5.1.7}) and
+ * unboxing (JLS {@jls 5.1.8}) conversions.
+ * These classes also provide
  * a number of methods for converting among primitive values, as well
- * as supporting such standard methods as equals and hashCode.  The
- * {@code Void} class is a non-instantiable class that holds a
- * reference to a {@code Class} object representing the type void.
+ * as supporting such standard methods as {@code equals} and {@code hashCode}.  The
```

Comment:
While the bulk of this sentence has unchanged, it read weirdly because of it's non-parallel structure:

"These classes ... provide ... methods for converting ..., as well as supporting ... methods ... equals and hashCode."

Consider these alternatives:

1. "These classes ... provide ... methods for converting ..., as well as support ... methods ... equals and hashCode."

2. "These classes ... provide ... methods for converting ..., as well as for supporting ... functionality of equals and hashCode."